gms/gossiper.hh

```diff
@@ -215,6 +215,7 @@ private:
 
     /* map where key is the endpoint and value is the state associated with the endpoint */
     std::unordered_map<inet_address, endpoint_state_ptr> _endpoint_state_map;
+    std::unordered_map<locator::host_id, endpoint_state_ptr> _endpoint_state_map_by_host_id;
```

Comment:
We should be able to have a single map indexed by host_id. If you don't have a host id, you're not allowed to insert states. That will simplify things.